clients/js/test/defaultGuards/token2022Payment.test.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  fetchToken,
+  setComputeUnitLimit,
+} from '@metaplex-foundation/mpl-essentials';
+import {
+  base58PublicKey,
+  generateSigner,
+  publicKey,
+  some,
+  transactionBuilder,
+} from '@metaplex-foundation/umi';
+import test from 'ava';
+import { mintV2 } from '../../src';
+import {
+  assertSuccessfulMint,
+  createCollectionNft,
+  createUmi,
+  createV2,
+} from '../_setup';
+import {
+  createMint,
+  getOrCreateAssociatedTokenAccount,
+  mintTo,
+} from '@solana/spl-token';
+import { Connection, LAMPORTS_PER_SOL, Signer } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+
+const SPL_TOKEN_2022 = new PublicKey(
+  'TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb'
+);
+
+test('it transfers Token2022 tokens from the payer to the destination', async (t) => {
+  const umi = await createUmi();
+  const connection = new Connection(umi.rpc.getEndpoint(), 'confirmed');
+
+  // Given a mint account such that:
+  // - The destination treasury has 100 tokens.
+  // - The payer has 12 tokens.
+  const destination = generateSigner(umi);
+  const signature = await connection.requestAirdrop(
+    new PublicKey(base58PublicKey(destination.publicKey)),
+    10 * LAMPORTS_PER_SOL
+  );
+  await connection.confirmTransaction(signature);
+
+  const destinationPayer = <Signer>{
+    publicKey: new PublicKey(base58PublicKey(destination.publicKey)),
+    secretKey: destination.secretKey,
+  };
+  // SPL Token 2022 mint account
+  const tokenMint = await createMint(
```

Comment:
I'd like to try and use Umi for this in the tests via mpl essentials so people don't think you have to use the `spl-token` library when reading the tests. I'll tackle this today real quick.